sdk/src/keystore/backend/mem.rs

```diff
@@ -0,0 +1,274 @@
+//! In Memory Keystore Backend that supports different cryptographic key operations such as key generation, signing, and public key retrieval.
+
+#[cfg(not(feature = "std"))]
+use alloc::{collections::BTreeMap, sync::Arc};
+#[cfg(feature = "std")]
+use std::{collections::BTreeMap, sync::Arc};
+#[cfg(feature = "keystore-bls381")]
+use w3f_bls::SerializableToBytes;
+
+use parking_lot::RwLock;
+
+use crate::keystore::{bls381, ecdsa, ed25519, sr25519, Backend};
+
+/// The type alias for the In Memory `KeyMap`.
+type KeyMap<P, S> = Arc<RwLock<BTreeMap<P, S>>>;
+
+#[cfg(feature = "keystore-ed25519")]
+#[derive(Debug, Clone)]
+struct Ed25519PublicWrapper(ed25519::Public);
+
+#[cfg(feature = "keystore-ed25519")]
+impl PartialEq for Ed25519PublicWrapper {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.as_ref() == other.0.as_ref()
+    }
+}
+
+#[cfg(feature = "keystore-ed25519")]
+impl Eq for Ed25519PublicWrapper {}
+
+#[cfg(feature = "keystore-ed25519")]
+impl Ord for Ed25519PublicWrapper {
+    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+        self.0.as_ref().cmp(other.0.as_ref())
+    }
+}
+
+#[cfg(feature = "keystore-ed25519")]
+impl PartialOrd for Ed25519PublicWrapper {
+    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+#[cfg(feature = "keystore-bls381")]
+#[derive(Clone, PartialEq, Eq)]
+struct Bls381PublicWrapper(bls381::Public);
+
+#[cfg(feature = "keystore-bls381")]
+impl PartialOrd for Bls381PublicWrapper {
+    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+#[cfg(feature = "keystore-bls381")]
+impl Ord for Bls381PublicWrapper {
+    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+        self.0.to_bytes().cmp(&other.0.to_bytes())
+    }
+}
+
+/// The In Memory Keystore Backend. It stores everything in memory
```

Comment:
`In-Memory` again, not sure if it is a big deal. I just think it helps with readability.